### PR TITLE
drivers: codec: doc: Cleanup Doxygen comments

### PR DESCRIPTION
**Pull request by kartben**
Added a few missing brief descriptions + fixed existing ones to use proper Javadoc-style comments.

### CODE REVIEW

File: include/zephyr/audio/codec.h [hunk 1]
```diff
@@ -32,82 +32,83 @@ extern "C" {
...
 
 /**
- * Digital Audio Interface Configuration
+ * @brief Digital Audio Interface Configuration.
+ *
  * Configuration is dependent on DAI type
  */
 typedef union {
-	struct i2s_config i2s;	/* I2S configuration */
+	struct i2s_config i2s;	/**< I2S configuration */
```
Comment (marc-hb) [inline include/zephyr/audio/codec.h:93]: See above.

File: include/zephyr/audio/codec.h [hunk 2]
```diff
@@ -32,82 +32,83 @@ extern "C" {
...
  */
 typedef enum {
-	AUDIO_PCM_WIDTH_16_BITS	= 16,
-	AUDIO_PCM_WIDTH_20_BITS	= 20,
-	AUDIO_PCM_WIDTH_24_BITS	= 24,
-	AUDIO_PCM_WIDTH_32_BITS	= 32,
+	AUDIO_PCM_WIDTH_16_BITS	= 16,	/**< 16-bit sample width */
+	AUDIO_PCM_WIDTH_20_BITS	= 20,	/**< 20-bit sample width */
+	AUDIO_PCM_WIDTH_24_BITS	= 24,	/**< 24-bit sample width */
+	AUDIO_PCM_WIDTH_32_BITS	= 32,	/**< 32-bit sample width */
```
[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/audio/codec.h:52]: You can check https://docs.zephyrproject.org/latest/doxygen/html/codec_8h.html vs. https://builds.zephyrproject.io/zephyr/pr/60362/docs/doxygen/html/codec_8h.html.

I don't know about "useful", but it's mostly a matter of making sure that all API has _explicit_ Doxygen comments IMO

File: include/zephyr/audio/codec.h [hunk 3]
```diff
@@ -32,82 +32,83 @@ extern "C" {
...
-	AUDIO_CHANNEL_FRONT_CENTER,
-	AUDIO_CHANNEL_REAR_LEFT,
-	AUDIO_CHANNEL_REAR_RIGHT,
-	AUDIO_CHANNEL_REAR_CENTER,
-	AUDIO_CHANNEL_SIDE_LEFT,
-	AUDIO_CHANNEL_SIDE_RIGHT,
-	AUDIO_CHANNEL_ALL,
+	AUDIO_CHANNEL_FRONT_LEFT,	/**< Front left channel */
+	AUDIO_CHANNEL_FRONT_RIGHT,	/**< Front right channel */
+	AUDIO_CHANNEL_LFE,		/**< Low frequency effect channel */
```
[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/audio/codec.h:77]: https://builds.zephyrproject.io/zephyr/pr/60362/docs/doxygen/html/group__audio__codec__interface.html#ga966cf980ce24855830e24b5de4d884d6 vs. https://docs.zephyrproject.org/latest/doxygen/html/group__audio__codec__interface.html#ga966cf980ce24855830e24b5de4d884d6

File: include/zephyr/audio/codec.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/audio/codec.h:52]: See also #56855. Right now we are not using the default Doxygen option which would be to treat as a warning any entity missing explicit documentation, but in an ideal world I think we want every API to be documented, even for stuff that may sound redundant.

[the remark this answers is not in the document]
Comment (marc-hb) [reply include/zephyr/audio/codec.h:52]: Thanks for the HTML  links! Loud and clear, very useful.

So I indeed have a strong preference for the version that does not merely duplicate the names and forces the reader to waste time reading them all only to realize in the end that there was no information that wasn't already in the name.

This being said, if there's a undiscerning rule somewhere that must be followed then I don't care enough to go and try to get it fixed :-)

Thanks for the PR!

[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/audio/codec.h:52]: mmh it actually looks like documenting each enum might not actually be required (to your point, enums often being reasonably self-described), but it is for union and struct members. I initially started cleaning up this file due to the warnings for the structs and unions, and decided to also go for the enums while at it...

[the remark this answers is not in the document]
Comment (marc-hb) [reply include/zephyr/audio/codec.h:52]: Would you have a direct link to the rules/policy? Sorry for being both curious _and_ lazy :-)

[the remark this answers is not in the document]
Comment (kartben) [reply include/zephyr/audio/codec.h:52]: No "rules" per se, but just leaning towards making sure that everything as proper Doxygen comments to ensure easier navigation, etc. Also related to https://github.com/zephyrproject-rtos/zephyr/issues/56855 - right now Doxygen is pretty lax on flagging undocumented entities, but I'd like to get to a point where all APIs are fully documented.

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**